src/Workflow/AbstractWorkflow.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Randock\Ddd\Workflow;
+
+
+use Randock\Ddd\Workflow\Exception\WorkflowException;
+
+abstract class AbstractWorkflow
+{
+    /**
+     * @return array
+     */
+    abstract public static function getTransitions(): array;
+
+    /**
+     * @param string $place
+     * @param string $transiton
+     *
+     * @throws WorkflowException
+     *
+     * @return mixed
+     */
+    public static function apply(string $place, string $transiton)
+    {
+        $transitions = static::getTransitions();
+        static::validateSchemaTransitions($transitions);
+        static::validateTransition($transiton, $transitions);
+
+        if (in_array($place, $transitions[$transiton]['from'])) {
+            return $place;
+        }
+
+        throw new WorkflowException(
+            sprintf('The transition (%s) can not be applied from (%s)', $transiton, $place)
+        );
+    }
+
+    /**
+     * @param array $transitions
+     *
+     * @throws WorkflowException
+     */
+    private static function validateSchemaTransitions(array $transitions)
```

Comment:
i think we should call this guard, since all our "guards" throw exceptions and "validate" return an array with errors